homeassistant/components/climate/__init__.py

```diff
@@ -235,10 +235,17 @@ def aux_heat_set_service(service):
     def temperature_set_service(service):
         """Set temperature on the target climate devices."""
         target_climate = component.extract_from_service(service)
-        kwargs = service.data
+
         for climate in target_climate:
-            climate.set_temperature(**kwargs)
+            kwargs = {}
+            for value, temp in service.data.items():
+                kwargs[value] = convert_temperature(
```

Comment:
I don't think this works for keys and values that are not temperatures in service data, eg entities.